app/models/katello/host/subscription_facet.rb

```diff
@@ -298,11 +298,12 @@ def backend_update_needed?
       end
 
       def self.populate_fields_from_facts(host, parser, _type, _source_proxy)
+        return unless host.subscription_facet # skip method if the host is not a registered host
         has_convert2rhel = parser.facts.key?('conversions.env.CONVERT2RHEL_THROUGH_FOREMAN')
         # Add in custom convert2rhel fact if system was converted using convert2rhel through Katello
         # We want the value nil unless the custom fact is present otherwise we get a 0 in the database which if debugging
         # might make you think it was converted2rhel but not with satellite, that is why I have the tenary below.
-        facet = host.subscription_facet || host.build_subscription_facet
```

Comment:
You use `facet.save unless facet.new_record?` below -- now `new_record?` can never be true, right?

Reply:
But I also do not exactly understand the reasoning for the `unless new_record?` -- why didn't we want to store it in the past?